packages/camera/camera_avfoundation/ios/Classes/CameraPlugin.m

```diff
@@ -93,9 +90,17 @@ - (void)orientationChanged:(NSNotification *)note {
 }
 
 - (void)sendDeviceOrientation:(UIDeviceOrientation)orientation {
-  [_deviceEventMethodChannel
-      invokeMethod:@"orientation_changed"
-         arguments:@{@"orientation" : FLTGetStringForUIDeviceOrientation(orientation)}];
+  __weak typeof(self) weakSelf = self;
+  dispatch_async(dispatch_get_main_queue(), ^{
+    [weakSelf.globalEventAPI
```

Comment:
I vaguely remember there was an effort to make method channel thread safe. what's the progress on that and do we still need to manually dispatch to main in plugin code?

Reply:
The current state is that replies are thread-safe, but calls that start on the host side still have to be done from the main thread. I don't think there's active work on addressing that part (although I believe it's still something we do eventually want to do).